client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -345,4 +358,42 @@ public void close() {
             this.connectionPool.close();
         }
     }
+
+    public CompletableFuture<Long> getDistanceBetweenTwoStreamCuts(Stream stream, StreamCut fromStreamCut, StreamCut toStreamCut) {
+        final CompletableFuture<StreamSegmentSuccessors> unread;
+        final Map<Segment, Long> endPositions;
+        if (toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            CompletableFuture<StreamInfo> streamInfo = fetchStreamInfo(stream.getScope(), stream.getStreamName());
```

Comment:
I tried tracing the code, in case of UNBOUNDED streamcuts, it was just setting an empty map, hence added the above check and fetched head/tail streamcuts accordingly.
please correct if wrong.